src/CoreWCF.Http/tests/AsyncNetAdoptionTests.cs

```diff
@@ -29,26 +30,50 @@ public void OneWayPatternTest()
                 var httpBinding = ClientHelper.GetBufferedModeBinding();
                 var factory = new System.ServiceModel.ChannelFactory<ClientContract.IOneWayContract>(httpBinding,
                     new System.ServiceModel.EndpointAddress(new Uri("http://localhost:8080/OneWayPatternTest/basichttp.svc")));
-                var channel = factory.CreateChannel();                              
-                Task task = channel.OneWay("Hello");
+                var channel = factory.CreateChannel();
+                await channel.OneWay("Hello");
+                Assert.Contains("Hello", host.Services.GetService<ConcurrentBag<string>>());
             }
-        }      
+        }
+
+        [Fact]
+        public async Task OneWayPatternTest_Parallel()
+        {
+            var inputs = Enumerable.Range(0, 100).Select(x => x.ToString()).ToArray();
+            var host = ServiceHelper.CreateWebHostBuilder<AsyncNetAdoptionOneWayServiceStartup>(_output).Build();
+            using (host)
+            {
+                host.Start();
+                var httpBinding = ClientHelper.GetBufferedModeBinding();
+                var factory = new System.ServiceModel.ChannelFactory<ClientContract.IOneWayContract>(httpBinding,
+                    new System.ServiceModel.EndpointAddress(new Uri("http://localhost:8080/OneWayPatternTest/basichttp.svc")));
+                var channel = factory.CreateChannel();
+                var tasks = inputs.AsParallel().Select(x => channel.OneWay(x)).ToList();
+                await Task.WhenAll(tasks);
+                foreach (string input in inputs)
+                {
+                    Assert.Contains(input, host.Services.GetService<ConcurrentBag<string>>());
+                }
```

Comment:
You still have a potential race here. It's unlikely to happen, but the task created in the service which adds the item to the concurrent bag might not execute before this validation step runs. In other words you might still have an entry missing. It would require the CPU to be heavily loaded and be running on a low core count machine, but the CI machines tend to fit this criteria (2 cores, highly contended CPU).